pkg/controller/controllerutil/controllerutil_test.go

```diff
@@ -256,6 +255,98 @@ var _ = Describe("Controllerutil", func() {
+			Expect(controllerutil.RemoveControllerReference(dep2, rs, scheme.Scheme)).To(HaveOccurred())
+		})
+		It("should only delete the controller reference and not the other owner references", func() {
+			rs := &appsv1.ReplicaSet{}
+			dep := &extensionsv1beta1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{Name: "foo", UID: "foo-uid"},
+			}
+			dep2 := &extensionsv1beta1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{Name: "bar", UID: "bar-uid"},
+			}
+			Expect(controllerutil.SetControllerReference(dep, rs, scheme.Scheme)).NotTo(HaveOccurred())
+			Expect(controllerutil.SetOwnerReference(dep2, rs, scheme.Scheme)).NotTo(HaveOccurred())
+			Expect(len(rs.GetOwnerReferences())).To(BeEquivalentTo(2))
+			Expect(controllerutil.RemoveControllerReference(dep, rs, scheme.Scheme)).NotTo(HaveOccurred())
+			Expect(len(rs.GetOwnerReferences())).To(BeEquivalentTo(1))
+		})
+		It("should only delete the controller reference and not the other owner references in different order", func() {
+			rs := &appsv1.ReplicaSet{}
+			dep := &extensionsv1beta1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{Name: "foo", UID: "foo-uid"},
+			}
+			dep2 := &extensionsv1beta1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{Name: "bar", UID: "bar-uid"},
+			}
+			Expect(controllerutil.SetOwnerReference(dep2, rs, scheme.Scheme)).NotTo(HaveOccurred())
+			Expect(controllerutil.SetControllerReference(dep, rs, scheme.Scheme)).NotTo(HaveOccurred())
+			Expect(len(rs.GetOwnerReferences())).To(BeEquivalentTo(2))
+			Expect(controllerutil.RemoveControllerReference(dep, rs, scheme.Scheme)).NotTo(HaveOccurred())
+			Expect(len(rs.GetOwnerReferences())).To(BeEquivalentTo(1))
+		})
+		It("should only fail because the scheme is wrong for the object", func() {
+			rs := &appsv1.ReplicaSet{}
+			dep := &extensionsv1beta1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{Name: "foo", UID: "foo-uid"},
+			}
+			dep2 := &extensionsv1beta1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{Name: "bar", UID: "bar-uid"},
+			}
+			Expect(controllerutil.SetOwnerReference(dep2, rs, scheme.Scheme)).NotTo(HaveOccurred())
+			Expect(controllerutil.SetControllerReference(dep, rs, scheme.Scheme)).NotTo(HaveOccurred())
+			Expect(len(rs.GetOwnerReferences())).To(BeEquivalentTo(2))
+			Expect(controllerutil.RemoveControllerReference(dep, rs, runtime.NewScheme())).To(HaveOccurred())
+			Expect(len(rs.GetOwnerReferences())).To(BeEquivalentTo(2))
+		})
+		It("should only fail because the object is not a runtime.Object", func() {
+			var obj metav1.Object
+			rs := &appsv1.ReplicaSet{}
+			dep := &extensionsv1beta1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{Name: "foo", UID: "foo-uid"},
+			}
+			Expect(controllerutil.SetControllerReference(dep, rs, scheme.Scheme)).NotTo(HaveOccurred())
+			Expect(len(rs.GetOwnerReferences())).To(BeEquivalentTo(1))
+			Expect(controllerutil.RemoveControllerReference(obj, rs, runtime.NewScheme())).To(HaveOccurred())
```

Comment:
I am going to change this to use the actual scheme and not the new scheme that is blank.

Reply:
Resolving, updated on latest push with the scheme adjusted to show that the error is coming from the object and not the scheme in function.